mesonbuild/coredata.py

```diff
@@ -848,13 +848,17 @@ def emit_base_options_warnings(self, enabled_opts: list):
             mlog.warning('Please see https://mesonbuild.com/Builtin-options.html#Notes_about_Apple_Bitcode_support for more details.', fatal=False)
 
 class CmdLineFileParser(configparser.ConfigParser):
-    def __init__(self):
+    def __init__(self) -> None:
         # We don't want ':' as key delimiter, otherwise it would break when
         # storing subproject options like "subproject:option=value"
         super().__init__(delimiters=['='], interpolation=None)
 
+    def optionxform(self, option: str) -> str:
+        # Don't call str.lower() on keys
+        return option
+
```

Comment:
Strictly speaking making keys case sensitive is not backward compatible, but I don't think it's a big deal in this case. However, I'm wondering if we should do the opposite and make optinterpreter case insensitive. I don't think it is desired to have 2 identical options that differ only by case. We already discussed to also normalize - to _ there.

Reply:
I thought about that too, and didn't know which was better, since that was also a backwards incompatible change. It would also mean making get_option case insensitive.

Reply:
There are cases where you want case sensitive variables in machine files. Specifically, for specifying (case sensitive) CMake variables in the machine files that are then forwarded to CMake. This is required / very annoying to work around otherwise.

Reply:
ok, LGTM then, we can always change options to be case insensitive later.